R/processes.R

```diff
@@ -263,39 +263,28 @@ create_event_based_processes <- function(
 
 #' @title Create processes for simulation
 #'
-#' @param psq model parameter
-#' @param pop population information
-#' @param max_age maximum age, defailt 100
-#'
-#' @return processes
+#' @param individuals individuals for model
+#' @param states states for model
+#' @param events events for model
+#' @param variables variables for model
+#' @return list of processes
 create_processes <- function(
-   psq,
-   pop,
-   max_age = 100) {
-
-   states <- create_states(psq)
-   variables <- create_variables(pop, max_age)
-   events <- create_events()
-   individual <- create_human(states, variables, events)
-
-   statesnamevector <- vector()
+  individuals,
+  states,
+  events,
+  variables
+) {
 
-   i <- 1
-   for (state in states) {
-      statesnamevector[i] <- state$name
-      i <- i + 1
-   }
+  processes <- list(
 
-   # Rendering process
-   process_render <- individual::state_count_renderer_process(
-      individual$name,
-      statesnamevector
-   )
+    infection_process(individuals, states, variables, events),
 
-   processes <- list(
-      process_render
-   )
+    individual::state_count_renderer_process(
+      individuals$name,
+      unlist(lapply(states, "[[", "name"))
```

Comment:
```suggestion
      vapply(states, "[[", FUN.VALUE = character(1), "name")
```

Reply:
Is there a reason for this change? It looks empirically that is not needed, and I do not see where the `vcapply` version would fail where that would **not** be the desired outcome